lib/rubocop/cop/util.rb

```diff
@@ -121,17 +121,25 @@ def to_supported_styles(enforced_style)
           .sub('Style', 'Styles')
       end
 
-      def tokens(node)
+      def tokens(node) # rubocop:disable Metrics/AbcSize
         @tokens ||= {}
         return @tokens[node.object_id] if @tokens[node.object_id]
 
         source_range = node.source_range
         begin_pos = source_range.begin_pos
         end_pos = source_range.end_pos
 
-        @tokens[node.object_id] = processed_source.tokens.select do |token|
-          token.end_pos <= end_pos && token.begin_pos >= begin_pos
+        tokens_to_node_end = processed_source.tokens.take_while do |token|
+          token.end_pos <= end_pos
```

Comment:
[take_while](http://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-take_while) will stop iterating once the condition returns `false` or `nil`. That means many less `token.end_pos` & `<=` operations.